entity-service/src/main/kotlin/com/egm/stellio/entity/web/EntityHandler.kt

```diff
@@ -119,72 +132,79 @@ class EntityHandler(
             .map {
                 entityService.appendEntityAttributes(entityId, it, disallowOverwrite)
             }
-            .flatMap {
+            .map {
                 logger.debug("Appended $it attributes on entity $entityId")
                 if (it.notUpdated.isEmpty())
-                    status(HttpStatus.NO_CONTENT).build()
+                    ResponseEntity.status(HttpStatus.NO_CONTENT).build<String>()
                 else
-                    status(HttpStatus.MULTI_STATUS).body(BodyInserters.fromValue(it))
+                    ResponseEntity.status(HttpStatus.MULTI_STATUS).body(it)
             }
     }
 
     /**
      * Implements 6.6.3.2 - Update Entity Attributes
      *
      */
-    fun updateEntityAttributes(req: ServerRequest): Mono<ServerResponse> {
-        val uri = req.pathVariable("entityId")
-        val contextLink = extractContextFromLinkHeader(req)
-
-        return req.bodyToMono<String>()
+    @PatchMapping("/{entityId}/attrs", consumes = [MediaType.APPLICATION_JSON_VALUE, JSON_LD_CONTENT_TYPE, JSON_MERGE_PATCH_CONTENT_TYPE])
+    fun updateEntityAttributes(
+        @RequestHeader httpHeaders: HttpHeaders,
+        @PathVariable entityId: String,
+        @RequestBody body: Mono<String>
+    ): Mono<ResponseEntity<*>> {
+        val contextLink = extractContextFromLinkHeader(httpHeaders.getOrEmpty("Link"))
+
+        return body
             .doOnNext {
-                if (!entityService.exists(uri)) throw ResourceNotFoundException("Entity $uri does not exist")
+                if (!entityService.exists(entityId)) throw ResourceNotFoundException("Entity $entityId does not exist")

```

Comment:
Not related to this PR too but this could be in a private method so we don't need to modify multiple lines when changing a variable name